develop/tutorials/articles/470-javascript/05-using-single-page-applications.markdown

```diff
@@ -0,0 +1,289 @@
+cleared. A zero value means the cache should never expire during SPA navigation. 
+A negative value means the cache should be disabled.
+
+**Request Timeout Time**: The time, in milliseconds, in which a SPA request 
+times out. A zero value means the request should never timeout.
+
+**User Notification Time**: The time, in milliseconds, in which a notification 
+is shown to the user stating that the request is taking longer than expected. A 
+zero value means no notification should be shown.
+
+Now that you know how to configure system settings for SPA, you can learn how to
+disable SPA for elements in your site next.
+
+### Disabling SPA
+
+Certain elements of your page may require a page refresh to work properly. For 
+example, you may have downloadable content that you want to share with the user. 
+In these cases, SPA must be disabled for those specific elements.
+
+If there is a certain portlet or page that you don't want to be part of the SPA, 
+and instead require a full page navigation, you have some options:
+
+-  Blacklist the portlet to disable SPA for the entire portlet
+-  Use the `data-senna-off` annotation to disable SPA for a specific link
+
+<!-- How do you disable SPA for a page? Since it's mentioned as an option, we need to demonstrate it. -->
+
+To blacklist a portlet from SPA follow these steps:
+
+1. Open your portlet's main class.
+
+2. Set the `_singlePageApplication` property to false:
+
+        _singlePageApplication = false;
+
+3. Alternatively, you can override the [`isSinglePageApplication` method](https://docs.liferay.com/ce/portal/7.0/javadocs/portal-impl/com/liferay/portal/model/impl/PortletImpl.html#isSinglePageApplication--)
+   of the portlet to return `false`.
+
+To disable SPA for a link follow these steps:
+
+1. Add the `data-senna-off` attribute to the link.
+
+2. Set the value to `true`.
+
+For example `<a data-senna-off="true" href="/pages/page2.html">Page 2</a>`
+
+That's all you need to do to disable SPA in your app.
+
+Now that you know how to disable SPA, you can learn about the available SPA
+lifecycle events next.
+
+## Listening to SPA Lifecycle Events
+
+During development, you may need to know when navigation has started or stopped 
+in your SPA. Luckily SennaJS makes this easy by exposing lifecycle events that 
+represent state changes in the application.  The following events are available:
+
+**beforeNavigate**: Fires before navigation starts. This event passes a JSON
+object with the path to the content being navigated to and whether to update the
+history. Below is an example event payload:
+
+    { path: '/pages/page1.html', replaceHistory: false }
+
+**startNavigate**: Fires when navigation begins. Below is an example event 
+payload:
+
+    { form: '<form name="form"></form>', path: '/pages/page1.html', 
+    replaceHistory: false }
+
+**endNavigate**: Fired after the content has been retrieved and inserted onto 
+the page. This event passes the following JSON object:
```

Comment:
Can you explain the startNavigate and endNavigate payloads? I'm not sure how
to parse the form, for example event.form returns undefined in the alert. I figure this is just because it is an object.(I'm sure that is why) Can you give an example of using these last two events?

looking here: https://github.com/liferay/liferay-portal/blob/master/modules/apps/foundation/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/app/App.es.js#L79-L92
It seems form is possible if there is form data being sent, but not always...

thx!